src/include/indexlr-workers.h

```diff
@@ -338,16 +332,22 @@ MinimizeWorker::work()
 
 			auto minimizers = getMinimizers(hashes, w);
 
-			if (stlfr) {
-				std::regex right("^.*#+");
-				std::regex left("/.*");
-
-				read.barcode = std::regex_replace(read.id, right, "");
-				read.barcode = std::regex_replace(read.barcode, left, "");
+			if (read.id.find("BX:Z:") == std::string::npos) {
+				size_t rightPos = read.id.find('#');
+				if (rightPos == std::string::npos) {
+					ss << read.barcode;
+				} else {
+					size_t leftPos = read.id.find('/', rightPos + 1);
+					if (leftPos == std::string::npos) {
+						ss << read.barcode;
```

Comment:
`read.barcode` should be empty here. What's the purpose of this line?

Reply:
Before I implemented this, `ss << read.barcode` will execute regardless of it the barcode is empty so I thought I should keep that

Reply:
The line of code is necessary in order to output NA for reads that have no barcode in the read